phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -149,16 +152,16 @@ public static void setFailDataTableUpdatesForTesting(boolean fail) {
       private final int clientVersion;
       // The collection of index mutations that will be applied before the data table mutations. The empty column (i.e.,
       // the verified column) will have the value false ("unverified") on these mutations
-      private Collection<Pair<Mutation, byte[]>> preIndexUpdates = Collections.emptyList();
+      private ListMultimap<HTableInterfaceReference, Mutation> preIndexUpdates;
```

Comment:
Doesn't have to be this patch, but we should consider extracting this ListMultimap<HTableInterfaceReference, Mutation> into a class of some kind with named accessor methods so that code dealing with it can be at a higher level of abstraction / more self-documenting.

Reply:
Do we have a jira for this? @gjacoby126  If not, I ll file one.

Reply:
There's no JIRA that I know of, @swaroopak. Thanks!

Reply:
Yes, there is a JIRA on this. It is https://issues.apache.org/jira/browse/PHOENIX-5339